src/aquarium-optimized/Aquarium.cpp

```diff
@@ -115,32 +115,33 @@ Aquarium::~Aquarium() {
   delete mFactory;
 }
 
-BACKENDTYPE Aquarium::getBackendType(const std::string &backendPath) {
+std::pair<BACKENDTYPE, BACKENDTYPE> Aquarium::getBackendType(
+    const std::string &backendPath) {
   if (backendPath == "opengl") {
-    return BACKENDTYPE::BACKENDTYPEOPENGL;
+    return {BACKENDTYPE::BACKENDTYPEOPENGL, BACKENDTYPE::BACKENDTYPELAST};
   } else if (backendPath == "dawn_d3d12") {
 #if defined(OS_WIN)
-    return BACKENDTYPE::BACKENDTYPEDAWND3D12;
+    return {BACKENDTYPE::BACKENDTYPEDAWN, BACKENDTYPE::BACKENDTYPED3D12};
 #endif
   } else if (backendPath == "dawn_metal") {
 #if defined(OS_MACOSX) && !defined(OS_IOS)
-    return BACKENDTYPE::BACKENDTYPEDAWNMETAL;
+    return {BACKENDTYPE::BACKENDTYPEDAWN, BACKENDTYPE::BACKENDTYPEMETAL};
 #endif
   } else if (backendPath == "dawn_vulkan") {
 #if defined(OS_WIN) || (defined(OS_LINUX) && !defined(OS_CHROMEOS))
-    return BACKENDTYPE::BACKENDTYPEDAWNVULKAN;
+    return {BACKENDTYPE::BACKENDTYPEDAWN, BACKENDTYPE::BACKENDTYPEVULKAN};
 #endif
   } else if (backendPath == "angle") {
```

Comment:
For "angle", we need to handle it the same way as "dawn", and exit the program if the actual native backends (other than OpenGL) are not supported. So we should accept "angle_opengl", "angle_d3d11", "angle_vulkan" and so on here.

Reply:
This should be the prospect, but the realistic is that we need at least:

* A patch that turns the current hardcoded D3D11 path into a switch-case
* A patch similar to #115

But I can hardly trust such patches because the ANGLE is unbuildable at the moment.